### PR TITLE
Triton Changes

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description

Closes 

## By Submitting this PR I confirm:
- I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- When the PR is ready for review, new or existing tests cover these changes.
- When the PR is ready for review, the documentation is up to date with these changes.

### CODE REVIEW

**Review by tzemicheal — APPROVED** [review]

File: examples/data_loss_prevention/stages/gliner_triton.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+
+    @property
+    def model(self) -> "GLiNER":
+        """
+        Return the GLiNER model instance.
+        """
+        if self._model is None:
+            from gliner import GLiNER
+            self._model = GLiNER.from_pretrained(self._model_source_dir,
+                                                 local_files_only=True,
```
Comment (tzemicheal) [inline examples/data_loss_prevention/stages/gliner_triton.py:54]: Could we change into this
 ```
self._model = GLiNER.from_pretrained(self._model_source_dir,
                                                 local_files_only=True,
                                                 onnx_path=self._onnx_path,
                                                 map_location=self._map_location,
                                                 load_onnx_model=True)
```

File: examples/data_loss_prevention/stages/gliner_triton.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+        # We load the model locally to use its pre/post-processing functions.
+        # The actual heavy inference will be done on Triton.
+        self._model = None
+        self._model_source_dir = model_source_dir
+        self._onnx_path = onnx_path
+        self._map_location = map_location
+        self.triton_model_name = triton_model_name
+        self.gliner_threshold = gliner_threshold
+        self.labels_embeddings = torch.tensor([])
```
Comment (tzemicheal) [inline examples/data_loss_prevention/stages/gliner_triton.py:43]: Change the self.labels_embedding to load from file. The `labels_embeddings.pt` is saved in the model directory

```
    labels_data = torch.load(os.path.join(model_source_dir,
                                              "labels_embeddings.pt"))
    self.labels_embeddings = labels_data["embeddings"]
```

File: examples/data_loss_prevention/stages/gliner_triton.py [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            all_entities.append(entities)
+        return all_entities
+
+    def pre_process(self, texts, labels):
+        """
+        Pre-process the data for the ONNX model.
+        """
+        # === 1. PRE-PROCESSING ===
+        if self.labels_embeddings.numel() == 0:
+            self.labels_embeddings = self.model.encode_labels(labels)
```
Comment (tzemicheal) [inline examples/data_loss_prevention/stages/gliner_triton.py:97]: Remove these checks, since the embedding label is already preloaded

File: examples/data_loss_prevention/stages/gliner_processor.py [hunk 4]
```diff
@@ -74,22 +74,20 @@ def __init__(self,
...
-        self._map_location = map_location
-        self._model = None
         self._model_max_batch_size = config.model_max_batch_size
         self.source_column_name = source_column_name
         self._regex_col_prefix = regex_col_prefix
         self.context_window = context_window
         self.fallback = fallback
         self._cache_dir = cache_dir
         self._needed_columns['dlp_findings'] = TypeId.STRING
+        self.gliner_triton = GliNERTritonInference(model_source_dir=model_source_dir, map_location=map_location)
```
Comment (tzemicheal) [inline examples/data_loss_prevention/stages/gliner_processor.py:90]:
```
self.gliner_triton = GliNERTritonInference(model_source_dir=model_source_dir, map_location=map_location,
gliner_threshold=confidence_threshold
)
```

**Review by tzemicheal — COMMENTED** [review]
Added few changes to the two files

File: examples/data_loss_prevention/stages/gliner_triton.py (hunk 1 above)
Comment (tzemicheal) [reply examples/data_loss_prevention/stages/gliner_triton.py:54]: to make sure it load from onnx model
